### ISSUE TITLE
Error: Exception at call of "GetTypes" by New-NavContainer with Artifacts 

### ISSUE THREAD

**Issue by git-no** [microsoft/navcontainerhelper#1130]
`New-NavContainer @params` with artifact from https://bcartifacts.azureedge.net/onprem/16.1.12629.12805/ch breaks with error
**Exception at call of "GetTypes" with 0 Arguments.**

### Error Stack
```powershell
Creating new container of version 16.1
Using artifact from https://bcartifacts.azureedge.net/onprem/16.1.12629.12805/ch
NavContainerHelper is version 0.7.0.16
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Enterprise - 2004
Docker Client Version is 19.03.8
Docker Server Version is 19.03.8
Removing C:\ProgramData\NavContainerHelper\Extensions\BIER-NEXT
Using image mcr.microsoft.com/dynamicsnav:10.0.19041.329-generic
Disabling Health Check (always report healthy)
Downloading application artifact /onprem/16.1.12629.12805/ch
Retrying download...

Ausnahme beim Aufrufen von "GetTypes" mit 0 Argument(en):  "Mindestens ein Typ in der Assembly kann nicht geladen werden. Rufen Sie die LoaderExceptions-Eigenschaft ab, wenn Sie weitere Informationen benötigen."
Script stack trace:
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.16\Misc\Download-File.ps1: Zeile 51
bei Download-File, C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.16\Misc\Download-File.ps1: Zeile 51
bei Download-Artifacts, C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.16\Misc\Download-Artifacts.ps1: Zeile 87
bei New-NavContainer, C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.16\ContainerHandling\New-NavContainer.ps1: Zeile 689
...
```

URL is valid and file can be downloaded via Web Browser.

Tried also to fix it by using 

```powershell
Flush-ContainerHelperCache -cache bcartifacts
```
but it did not solve the issue.

Parameters are `-auth: NavUserPassword -shortcuts: CommonDesktopFolder -accept_eula: True -artifactUrl: https://bcartifacts.azureedge.net/onprem/16.1.12629.12805/ch -licenseFile: C:\temp\labs\BIER\UPG-16.1#NEXT\fin.flf -doNotCheckHealth: True -containerName: BIER-NEXT -useBestContainerOS: True -updateHosts: True -credential: System.Management.Automation.PSCredential -enableTaskScheduler: False -dumpEventLog: True`

**Comment by git-no** [comment]
Freddy, is the url used by file `https://github.com/microsoft/navcontainerhelper/blob/77caf97f39c9ffbb40f61c559edd677f80b0725d/Misc/Download-Artifacts.ps1#L49` in line 49 correct?

See my error stack line above `Downloading application artifact /onprem/16.1.12629.12805/ch`. This address seems not correct to me.

**Comment by freddydk** [comment]
The problem in the error is that there are dotnet objects in PowerShell which cannot be loaded.
I will change the code to not enumerate the types

**Comment by git-no** [comment]
I can confirm, it works with the change, with your version 0.7.0.17.
Tusind tak!

**Comment by freddydk** [comment]
I have actually made one more change - to add the dotnet type in the module main code - and not in Compile + download files - the problem was duplicate dotnet type loading, which I tried to solve using a test whether the type was added. That test apparently didn't work.

**Comment by freddydk** [comment]
Fixed in 0.7.0.17
